riklet/src/core.rs

```diff
@@ -236,22 +237,16 @@ impl Riklet {
             let tap_ip = subnet.nth(1).ok_or("Fail get tap ip")?;
             let firecracker_ip = subnet.nth(2).ok_or("Fail to get firecracker ip")?;
 
-            let output = Command::new("/bin/sh")
-                .arg(&self.function_config.script_path)
-                .arg(&workload_definition.name)
-                .arg(tap_ip.to_string())
-                .output()?;
-
-            if !output.status.success() {
-                let stderr = String::from_utf8_lossy(&output.stderr);
-                if !stderr.is_empty() {
-                    event!(Level::ERROR, "stderr: {}", stderr);
-                }
-                return Err(stderr.into());
-            }
+            let config = NetworkInterfaceConfig::new(
+                workload.instance_id.clone(),
+                workload_definition.name.clone(),
```

Comment:
We can secure this part of code. iface name has to be less than 16 characters to be valid. I think we can verify it here and use an other name for interface if workload name is too long.

Reply:
Checking the function name is not a good answer for this issue as the function is created upstream in the controller. 
That means that we will need to check the function length also in the controller.
It's better to find another identifier.

This work can be done in another Issue.

Reply:
I can add easily some validation around the name of the interface, it makes sense @mflagey.

> Checking the function name is not a good answer for this issue as the function is created upstream in the controller.

@kalil-pelissier We could define a format that is used for iface, it can be done here, that would also make sense IMO.

Reply:
It has been fixed by defining of pattern of iface, and the ability to just ask for randomly generated iface name which is a truncated version of the id.